# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/prim_store/text_run.rs
# Commit: b7f88e5c537b
# Full Hash: b7f88e5c537bb7d64ba9d4cb58b052650d510638
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-02-08 17:32:27
# Regressor Bug: 1749380
# File Overlap Count: 1
# Description:
#   Bug 1749380 - Improve how WR handles bounding rects for off-screen surfaces r=gfx-reviewers,kvark
#   
#   This patch introduces a number of subtle but important changes
#   to how we deal with off-screen surfaces. The overall goals are:
#   
# ==============================================================================

diff -r 8e9c1ffdeed8 -r b7f88e5c537b gfx/wr/webrender/src/prim_store/text_run.rs
--- a/gfx/wr/webrender/src/prim_store/text_run.rs	Mon Feb 07 22:02:35 2022 +0000
+++ b/gfx/wr/webrender/src/prim_store/text_run.rs	Mon Feb 07 22:14:47 2022 +0000
@@ -246,7 +246,6 @@
         transform: &LayoutToWorldTransform,
         mut allow_subpixel: bool,
         raster_space: RasterSpace,
-        root_scaling_factor: f32,
         spatial_tree: &SpatialTree,
     ) -> bool {
         // If local raster space is specified, include that in the scale
@@ -257,10 +256,7 @@
         //           will no longer be required.
         let raster_scale = raster_space.local_scale().unwrap_or(1.0).max(0.001);
 
-        // root_scaling_factor is used to scale very large pictures that establish
-        // a raster root back to something sane, thus scale the device size accordingly.
-        // to the shader it looks like a change in DPI which it already supports.
-        let dps = surface.device_pixel_scale.0 * root_scaling_factor;
+        let dps = surface.device_pixel_scale.0;
         let font_size = specified_font.size.to_f32_px();
 
         // Small floating point error can accumulate in the raster * device_pixel scale.
@@ -442,7 +438,6 @@
         transform: &LayoutToWorldTransform,
         surface: &SurfaceInfo,
         spatial_node_index: SpatialNodeIndex,
-        root_scaling_factor: f32,
         allow_subpixel: bool,
         low_quality_pinch_zoom: bool,
         resource_cache: &mut ResourceCache,
@@ -464,14 +459,13 @@
             transform,
             allow_subpixel,
             raster_space,
-            root_scaling_factor,
             spatial_tree,
         );
 
         if self.glyph_keys_range.is_empty() || cache_dirty {
             let subpx_dir = self.used_font.get_subpx_dir();
 
-            let dps = surface.device_pixel_scale.0 * root_scaling_factor;
+            let dps = surface.device_pixel_scale.0;
             let transform = match raster_space {
                 RasterSpace::Local(scale) => FontTransform::new(scale * dps, 0.0, 0.0, scale * dps),
                 RasterSpace::Screen => self.used_font.transform.scale(dps),